### PR TITLE
set the mask bit on ping responses (pongs)

### PR DESCRIPTION
**Pull request by rentallect**

### CODE REVIEW

File: src/websocket.c [hunk 1]
```diff
@@ -389,7 +389,7 @@ static void send_pong(um_websocket_t *ws, const char* ping_data, int len) {
     buf.base = malloc(buf.len);
 
     buf.base[0] = WS_FIN | OpCode_Pong;
-    buf.base[1] = (char)(0x7f & len);
+    buf.base[1] = (char)(WS_MASK | (0x7f & len));
     if (ping_data != NULL && len > 0) {
         memcpy(buf.base + 2, ping_data, len);
```
Comment (ekoby) [inline src/websocket.c:394]: don't you need to add mask bytes and mask ping data?

**Comment by ekoby** [comment]
I added HTTP keep-alive fix on this PR

File: src/websocket.c (hunk 1 above)
Comment (rentallect) [reply src/websocket.c:394]: Yes I do. The one-liner above was my sanity check on wether MM server would stop terminating the connection on ping/pongs, so I can continue chasing another issue.  You also said you wanted to get some tweaks into a PR, so I published this for you to pile on.

Comment (ekoby) [reply src/websocket.c:394]: my contribution to the pile is comlete

File: src/websocket.c [hunk 2]
```diff
@@ -383,15 +383,35 @@ void ws_read_cb(uv_link_t *l, ssize_t nread, const uv_buf_t *buf) {
...
+static void send_pong(um_websocket_t *ws, const char* ping_data, int len, bool masked, uint8_t mask[4]) {
     uv_buf_t buf;
     buf.len = 2 + len;
+    if (masked) {
+        buf.len += sizeof(mask);
+    }
     buf.base = malloc(buf.len);
 
     buf.base[0] = WS_FIN | OpCode_Pong;
     buf.base[1] = (char)(WS_MASK | (0x7f & len));
```
Comment (ekoby) [inline src/websocket.c:395]: `WS_MASK` flag is always set so `masked` parameter should also be always be true, right?

File: src/websocket.c [hunk 3]
```diff
@@ -370,8 +370,8 @@ void ws_read_cb(uv_link_t *l, ssize_t nread, const uv_buf_t *buf) {
             ws->read_cb((uv_stream_t *) ws, UV_EOF, buf);
             break;
         case OpCode_Ping:
-            UM_LOG(TRACE, "got ping");
-            send_pong(ws, dp, len);
+            UM_LOG(TRACE, "got ping masked=%d len=%d", masked, len);
+            send_pong(ws, dp, len, masked, mask);
```
Comment (ekoby) [inline src/websocket.c:374]: we should not use server masked flag here.
according to https://tools.ietf.org/id/draft-ietf-hybi-thewebsocketprotocol-09.html#rfc.section.4.3
all client frames have to be masked

**Pull request merged**